openpgp.d.ts

```diff
@@ -22,6 +22,7 @@ export function generateSessionKey(options: { encryptionKeys: PublicKey[], date?
 export function decryptKey(options: { privateKey: PrivateKey; passphrase?: string | string[]; config?: PartialConfig }): Promise<PrivateKey>;
 export function encryptKey(options: { privateKey: PrivateKey; passphrase?: string | string[]; config?: PartialConfig }): Promise<PrivateKey>;
 export function reformatKey(options: { privateKey: PrivateKey; userIDs?: UserID|UserID[]; passphrase?: string; keyExpirationTime?: number; config?: PartialConfig }): Promise<KeyPair>;
+export function revokeKey(options: {key: Key, revocationCertificate?: String, reasonForRevocation?: {flag?: enums.reasonForRevocation, string?: String}, config?: PartialConfig}): Promise<KeyPair|PublicKey>
```

Comment:
Hey :wave: Thanks for the PR! :+1:

A few nitpicks:
- Please try to follow the formatting of the surrounding code (spaces in object literals, simicolons);
- Since the `revocationCertificate` and `reasonForRevocation` parameters are essentially exclusive, could you create two overloads of this function, each with only one of them? That should make it clearer how this function can be used.
- And then finally, if you don't pass a revocation certificate, you have to pass a `PrivateKey`, whereas if you do, you can also pass a `PublicKey` (and the return type depends on whether you do - so you might in fact want three overloads - though perhaps we should just simplify the function at some point).

Reply:
Thank you for the feedback. I think this can be consolidated to 2 overloads unless I am forgetting something. One overload for when `PrivateKey` is passed and another when `PublicKey` is passed. Let me know what you think of this.